### PR TITLE
Add support of DateType and TimestampType for GetTimestamp expression

### PR DESCRIPTION
**Pull request by nartal1**
This is follow on PR for the review comment: https://github.com/NVIDIA/spark-rapids/pull/1742#discussion_r579487836

Signed-off-by: Niranjan Artal <nartal@nvidia.com>

### CODE REVIEW

**Comment by nartal1** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/catalyst/expressions/rapids/Timestamp.scala [hunk 1]
```diff
@@ -28,7 +28,9 @@ object TimeStamp {
     GpuOverrides.expr[GetTimestamp](
       "Gets timestamps from strings using given pattern.",
       ExprChecks.binaryProjectNotLambda(TypeSig.TIMESTAMP, TypeSig.TIMESTAMP,
-        ("timeExp", TypeSig.STRING, TypeSig.STRING),
+        ("timeExp",
+            TypeSig.STRING + TypeSig.DATE + TypeSig.TIMESTAMP,
```
Comment (revans2) [inline sql-plugin/src/main/scala/org/apache/spark/sql/catalyst/expressions/rapids/Timestamp.scala:32]: I would prefer to have us say we only support STRING unless we have tests to back up support for the other types as well.

**Comment by nartal1** [comment]
@revans2 Please take a look. I have addressed the review comment https://github.com/NVIDIA/spark-rapids/pull/1742#discussion_r579487836.

File: sql-plugin/src/main/scala/org/apache/spark/sql/catalyst/expressions/rapids/Timestamp.scala (hunk 1 above)
Comment (nartal1) [reply sql-plugin/src/main/scala/org/apache/spark/sql/catalyst/expressions/rapids/Timestamp.scala:32]: Added tests for date and timestamp type. PTAL.

**Review by revans2 — APPROVED** [review]

**Comment by revans2** [comment]
build

**Pull request merged**